### PR TITLE
idea for how to handle transactions

### PR DESCRIPTION
**Pull request by akeiper**

### CODE REVIEW

File: Contract/Savable.php [hunk 1]
```diff
@@ -3,5 +3,6 @@
 
 interface Savable {
 
+	//@question: is by user always required?
```
Comment (bgarb002) [inline Contract/Savable.php:6]: Yes

File: Service/UserDiscountCreate.php [hunk 2]
```diff
@@ -1,23 +1,84 @@
...
+	public function commit(): void
+	{
+		foreach ($this->repos as $r) {
+			if ($r instanceof  Transactional) {
+				$r->commit();
+			}
+		}
 	}
+
+	public function rollback(): void
```
Comment (bgarb002) [inline Service/UserDiscountCreate.php]: Ok move these 3 functions to an abstract command class that has execute as an abstract method.

File: Service/UserDiscountCreate.php [hunk 3]
```diff
@@ -1,23 +1,84 @@
...
 namespace Zumba\User\Service;
 
-use \Zumba\User\Contract\Command;
+use \Zumba\User\Contract\Command,
+	\Zumba\User\Factory\EventQueueRepo,
+	\Zumba\User\Factory\Repo,
+	\Zumba\User\Contract\Transactional;
 
-class UserDiscountCreate implements Command {
+class UserDiscountCreate implements Command, Transactional {
```
Comment (bgarb002) [inline Service/UserDiscountCreate.php]: This would extend abstract command. The parent class will implement Transactional.

**Comment by akeiper** [comment]
@bgarb002 moved to an abstract class. Let me know what you guys think

File: Service/Command.php [hunk 4]
```diff
@@ -0,0 +1,51 @@
+<?php
+namespace Zumba\User\Service;
+
+use \Zumba\User\Contract\Command as CommandInterface,
+	\Zumba\User\Contract\Transactional;
+
+abstract class Command implements CommandInterface, Transactional {
```
Comment (bgarb002) [inline Service/Command.php:7]: Where is the definition of CommandInterface?

Comment (akeiper) [reply Service/Command.php:7]: @bgarb002 

```use \Zumba\User\Contract\Command as CommandInterface,```

**Review by bgarb002 — APPROVED** [review]

**Review by ZurabWeb — APPROVED** [review]

**Pull request merged**